applications/tari_validator_node/src/p2p/services/epoch_manager/base_layer_epoch_manager.rs

```diff
@@ -94,9 +106,42 @@ impl BaseLayerEpochManager {
 
         // If the committee size is bigger than vns.len() then this function is broken.
         let mut base_node_client = self.base_node_client.clone();
-        let mut vns = base_node_client.get_validator_nodes(epoch.0 * 10).await?;
+        let mut vns = base_node_client.get_validator_nodes(epoch.to_height()).await?;
         vns.sort_by(|a, b| a.shard_key.partial_cmp(&b.shard_key).unwrap());
 
+        let current_shard_key = vns
```

Comment:
Logic looks like it's on the right track, but design-wise, I would prefer a separate service, `SyncManagerService`, that listens for a `broadcast` event on a `channel` and then runs this logic. Finally, the epoch manager can publish that event here

Reply:
Perfect, I really like it!